app/controllers/api/functions.php

```diff
@@ -474,6 +475,38 @@
         ])));
 
         if ($next) {
+            $schedule = Authorization::skip(function () use ($dbForConsole, $project, $function) {
+                return $dbForConsole->findOne('schedules', [
+                    Query::equal('region', [App::getEnv('_APP_REGION')]), // Todo replace with projects region
+                    Query::equal('type', ['function']),
+                    Query::equal('projectId', [$project->getId()]),
+                    Query::equal('scheduleId', [$function->getId()]),
+                ]);
+            });
+
+            // TODO constrain with unique index ??
+            if (empty($schedule)) {
+                Authorization::skip(
+                    fn() => $dbForConsole->createDocument('schedules', new Document([
+                        'region' => App::getEnv('_APP_REGION'), // Todo replace with projects region
+                        'type' => 'function',
+                        'scheduleId' => $function->getId(),
+                        'projectId' => $project->getId(),
+                        'scheduleUpdatedAt' => $function['scheduleUpdatedAt'],
```

Comment:
We should use `getAttribute` to get documents properties. Its safer in terms of null safety check is not needed with it, and it provides the ability to return proper default value.